app/src/pages/viewPostPage/viewPost.jsx

```diff
@@ -0,0 +1,173 @@
+import styles from "./viewPost.module.css"
+import Header from "../../components/header/header"
+import Footer from "../../components/footer/footer"
+import PostCmp from '../../components/post/post'
+
+import API from "../../utils/api";
+
+import { useEffect, useState } from "react";
+import { Navigate, useLocation, useParams } from "react-router-dom";
+import GenImage from "../../components/genImage/GenImage";
+
+import Cookies from "universal-cookie";
+
+function Bid(props){
+    const cookies = new Cookies("/", {path : null})
+    const currentUserId = cookies.get("currentUser").id
+
+    const {description, images} = props;
+    const bid = {...props.bid, ...{'posterId': currentUserId}}
+    const commissioned = bid.commissionId ? true : false
+
+
+    return (
+        <div className={styles.bid} >
+            <div className={styles.bidImgs}>
+                { (images.length > 0) && images.map((image)=> (
+                    <img src={image.imgUrl} />
+                ))}
+            </div>
+            <p>{description}</p>
+
+            {
+                (commissioned) && (
+                    <p> Commissioned! </p>
+                )
+            }
+
+            {
+                (!commissioned) && (
+                    <button
+                    onClick={()=> {API.addCommission(bid)}}
+                    >Commision</button>
+                )
+            }
+
+        </div>
+    )
+}
+
+export default function ViewPost(){
+
+    // TODO: use cookies to minimize loading time
+    const postId = useParams().id;
+    const cookies = new Cookies("/", {path : null})
+    const currentUserId = cookies.get("currentUser").id
+
+    const [postData, setPostData] = useState({})
+    const [imageIdxs, setImageIdxs] = useState([0, 1, 2])
+
+    const [navBid, setNavBid] = useState(false)
+
+    useEffect(() => {
+        async function getData(id){
+            const data = await API.getPost(id);
+            setPostData(data)
+        }
+
+        getData(postId)
+    }, [])
+
+
+    // TODO: Has to be a better way to implement this
+    // TODO: Add bidder information and ratings
+    if (postData.userId == currentUserId){
```

Comment:
Thanks for handling the view for the person who the post and other viewers.

A way to make the code cleaner is to create a different component `ViewPostAsOwner` for the owner of the post and call that here! and we can also separate the default view into something like `ViewPostAsNonOwner` or something like that. It's flexible but I appreciate you planned for the different views

Reply:
They can also share components within each of them if you want to modularize it even more (such as the image display).  This is not a blocking change but just some ideas!